webapp/src/compiler.ts

```diff
@@ -360,7 +360,7 @@ function pyDecompileCoreAsync(opts: pxtc.CompileOptions, fileName: string): Prom
     return workerOpAsync("pydecompile", { options: opts, fileName: fileName })
 }
 
-export function workerOpAsync(op: string, arg: pxtc.service.OpArg) {
+export function workerOpAsync<T extends keyof pxtc.service.ServiceOps>(op: T, arg: pxtc.service.OpArg): Promise<any> {
```

Comment:
should the return type be `Promise< OpResOrError >`?

Reply:
Yes, but it would require more refactoring and I've already done a fair bit in this PR. This type is more precise than it was and we can improve again.